lib/ansible/module_utils/vmware.py

```diff
@@ -813,13 +813,69 @@ def __init__(self, module):
         self.current_vm_obj = None
         self.content = connect_to_api(self.module)
 
+    def get_managed_objects_properties(self, vim_type, properties=None):
+        """
+        Function to look up a Managed Object Reference in vCenter / ESXi Environment
+        :param vim_type: Type of vim object e.g, for datacenter - vim.Datacenter
+        :param properties: List of properties related to vim object e.g. Name
+        :return: local content object
+        """
+        # Get Root Folder
+        root_folder = self.content.rootFolder
+
+        if properties is None:
+            properties = ['name']
+
+        # Create Container View with default root folder
+        mor = self.content.viewManager.CreateContainerView(root_folder, [vim_type], True)
+
+        # Create Traversal spec
+        traversal_spec = vmodl.query.PropertyCollector.TraversalSpec(
+            name="traversal_spec",
+            path='view',
+            skip=False,
+            type=vim.view.ContainerView
+        )
+
+        # Create Property Spec
+        property_spec = vmodl.query.PropertyCollector.PropertySpec(
+            type=vim_type,  # Type of object to retrieved
+            all=False,
+            pathSet=properties
+        )
+
+        # Create Object Spec
+        object_spec = vmodl.query.PropertyCollector.ObjectSpec(
+            obj=mor,
+            skip=True,
+            selectSet=[traversal_spec]
+        )
+
+        # Create Filter Spec
+        filter_spec = vmodl.query.PropertyCollector.FilterSpec(
+            objectSet=[object_spec],
+            propSet=[property_spec],
+            reportMissingObjectsInResults=False
+        )
+
+        return self.content.propertyCollector.RetrieveContents([filter_spec])
+
     def get_vm(self):
         vm = None
         match_first = (self.params['name_match'] == 'first')
 
         if self.params['uuid']:
             vm = find_vm_by_id(self.content, vm_id=self.params['uuid'], vm_id_type="uuid")
-        elif self.params['folder'] and self.params['name']:
+
+        elif self.params['name']:
+            vms = self.get_managed_objects_properties(vim_type=vim.VirtualMachine, properties=['name'])
+            for vm_obj in vms:
+                if self.params['name'] == vm_obj.obj.name:
+                    vm = vm_obj.obj
```

Comment:
If you check for the datacenter name and folder path here (before breaking) you could return the result and ignore all the other strategies.

Reply:
OK. I add this.